platform/admin/configurator/admin-configurator-api/src/main/java/org/codice/ddf/admin/configurator/Status.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public enum Status {
+    COMMIT_PASSED, COMMIT_FAILED, SKIPPED, ROLLBACK_PASSED, ROLLBACK_FAILED;
```

Comment:
❓What sequence of events could cause an operation to get in the `COMMIT_FAILED` state and remain there?
❓Instead of using an `enum`, could we create sub-classes of `Result` and `ConfigurationException` to capture an operation's result and reason for failure? This would allow operations to create new exceptions as needed and add more useful information regarding the failure such as a factory PID, file path, etc.